contributors/design-proposals/admission_control_extension.md

```diff
@@ -0,0 +1,594 @@
+configuration that may vary by resource type. If initializers is set the admission controller will check
+whether the user has the ability to run the `initialize` verb on the current resource type, and reject
+the entry with 403 if not. This allows a privileged user to bypass initialization by setting `initializers`
+to the empty struct.
+
+Once created, an object is not visible to clients unless the following conditions are satisfied:
+
+1. The initializers field is null.
+2. The client provides a special option to GET, LIST, or WATCH indicating that the user wishes to see uninitialized objects.
+
+The apiserver that accepts the incoming creation should hold the response until the object is
+initialized or the timeout is exceeded. This increases latency, but allows clients to avoid breaking
+semantics. If the apiserver reaches the timeout it must return an appropriate error that includes the
+resource version of the object and UID so that clients can perform a watch. If an initializer reports
+a `Result` with a failure, it must return that to the user (all failures result in deletion).
+
+Each initializer is a controller or other client agent that watches for new objects with an initializer
+whose first position matches their assigned name (e.g. `PodAutoSizer`) and then operate on them. These
+clients would use the `?includeUninitialized=true` query param (working name) and observe all
+objects.
+
+The initializer would perform a normal update on the object to perform their function, and then
+remove their entry from `initializers` (or adding more entries). If an error occurs during initialization
+that must terminate initialization, the `Status` field on the initializer should be set instead of removing
+the initializer entry and then the initializer should delete the object.
+
+During initialization, resources may have relaxed validation requirements, which means initializers must
+handle incomplete objects. The create call will perform normal defaulting so that initializers are not providing
+their own defaulting, including UID and creationTimestamp. At all phases the object must be valid, so
+resources that wish to use initializers should consider how defaulting would complicate initializers.
+
+To allow naive clients to avoid having to deal with uninitialized objects, the API will automatically
+filter uninitialized objects out LIST and WATCH. Explicit GETs to that object should return the
+appropriate status code `202 Accepted` indicating that the resource is reserved and including a `Status`
+response with the correct resource version, but not the object. Clients specifying
+`includeUninitialized` will see all updates, but shared code like caches and informers may need
+to implement layered filters to handle multiple clients requesting both variants. A CREATE to an
+uninitialized object should report the same status as before, and DELETE is always allowed.
```

Comment:
Yes, will clarify.  Update should probably be Forbidden, because some users will be allowed to update prior to creation.  We may want to add a sub cause to the error that indicates that this is a temporary condition (we may want todo that in general).  Most clients have to deal with Forbidden, so I'm not as worried about that.